x-pack/plugin/blob-cache/src/main/java/org/elasticsearch/blobcache/shared/SharedBlobCacheService.java

```diff
@@ -472,14 +472,18 @@ private void assignToSlot(Entry<CacheFileRegion> entry, int freeSlot) {
                 assert regionOwners.compareAndSet(freeSlot, entry.chunk, null);
                 freeRegions.add(freeSlot);
                 keyMapping.remove(entry.chunk.regionKey, entry);
-                throw new AlreadyClosedException("evicted during free region allocation");
+                throwAlreadyClosed("evicted during free region allocation");
             }
             pushEntryToBack(entry);
             // assign sharedBytesPos only when chunk is ready for use. Under lock to avoid concurrent tryEvict.
             entry.chunk.sharedBytesPos = freeSlot;
         }
     }
 
+    private static void throwAlreadyClosed(String message) {
+        throw new AlreadyClosedException(message);
+    }
```

Comment:
But does this actually save anything?

Reply:
If this was the cold path yes a tiny bit, but that's actually not really the point here. I want to get rid of using these exceptions for flow control (they are caught way up the stack and translated into uncached reads) to both dry up the code (we do this separately in frozen/cold and stateless right now) and speed things up.